src/app/layout/qgslayoutdesignerdialog.cpp

```diff
@@ -1408,6 +1416,234 @@ void QgsLayoutDesignerDialog::deleteLayout()
+  imageDlg.setGenerateWorldFile( mLayout->customProperty( QStringLiteral( "exportWorldFile" ), false ).toBool() );
+  imageDlg.setAntialiasing( antialias );
+
+#if 0 //TODO
+  QgsAtlasComposition *atlasMap = &mComposition->atlasComposition();
+#endif
+
+  QString outputFileName;
+#if 0 //TODO
+  if ( atlasMap->enabled() && mComposition->atlasMode() == QgsComposition::PreviewAtlas )
+  {
+    QString lastUsedDir = settings.value( QStringLiteral( "UI/lastSaveAsImageDir" ), QDir::homePath() ).toString();
+    outputFileName = QDir( lastUsedDir ).filePath( atlasMap->currentFilename() );
+  }
+#endif
+
+#ifdef Q_OS_MAC
+  mQgis->activateWindow();
+  this->raise();
+#endif
+  QPair<QString, QString> fileNExt = QgsGuiUtils::getSaveAsImageName( this, tr( "Save layout as" ), outputFileName );
+  this->activateWindow();
+
+  if ( fileNExt.first.isEmpty() )
+  {
+    return;
+  }
+
+  if ( !imageDlg.exec() )
+    return;
+
+  cropToContents = imageDlg.cropToContents();
+  imageDlg.getCropMargins( marginTop, marginRight, marginBottom, marginLeft );
+  mLayout->setCustomProperty( QStringLiteral( "imageCropToContents" ), cropToContents );
+  mLayout->setCustomProperty( QStringLiteral( "imageCropMarginTop" ), marginTop );
+  mLayout->setCustomProperty( QStringLiteral( "imageCropMarginRight" ), marginRight );
+  mLayout->setCustomProperty( QStringLiteral( "imageCropMarginBottom" ), marginBottom );
+  mLayout->setCustomProperty( QStringLiteral( "imageCropMarginLeft" ), marginLeft );
+
+  mLayout->setCustomProperty( QStringLiteral( "imageAntialias" ), imageDlg.antialiasing() );
+
+  mView->setPaintingEnabled( false );
+  QApplication::setOverrideCursor( Qt::BusyCursor );
+
+  QgsLayoutExporter exporter( mLayout );
+
+  QgsLayoutExporter::ImageExportSettings settings;
+  settings.cropToContents = cropToContents;
+  settings.cropMargins = QgsMargins( marginLeft, marginTop, marginRight, marginBottom );
+  settings.dpi = imageDlg.resolution();
+  if ( hasUniformPageSizes )
+  {
+    settings.imageSize = QSize( imageDlg.imageWidth(), imageDlg.imageHeight() );
+  }
+  settings.generateWorldFile = imageDlg.generateWorldFile();
+  settings.flags = QgsLayoutContext::FlagUseAdvancedEffects;
+  if ( imageDlg.antialiasing() )
+    settings.flags |= QgsLayoutContext::FlagAntialiasing;
+
+  switch ( exporter.exportToImage( fileNExt.first, settings ) )
+  {
+    case QgsLayoutExporter::Success:
+    case QgsLayoutExporter::PrintError:
+      break;
+
+    case QgsLayoutExporter::FileError:
+      QMessageBox::warning( this, tr( "Image Export Error" ),
+                            tr( "Cannot write to %1.\n\nThis file may be open in another application." ).arg( exporter.errorFile() ),
+                            QMessageBox::Ok,
+                            QMessageBox::Ok );
+      break;
+
+    case QgsLayoutExporter::MemoryError:
+      QMessageBox::warning( nullptr, tr( "Memory Allocation Error" ),
```

Comment:
Is it wanted that the parent of `QMessageBox` is `this` in case of `FileError` and `nullptr` in case of `MemoryError`?

Reply:
good catch - it was a copy/paste from the old composer code. I'll add the parent to all these. Or @nirvn should I use the message bar here?

Reply:
Fixed